index.html

```diff
@@ -355,6 +359,14 @@ <h2><code>PointerEvent</code> Interface</h2>
                         <dd>
                             <p>The clockwise rotation (in degrees, in the range of [0,359]) of a transducer (e.g. pen stylus) around its own major axis. For hardware and platforms that do not report twist, the value MUST be 0.</p>
                         </dd>
+                    <dt><dfn>altitudeAngle</dfn></dt>
+                        <dd>
+                            <p>The altitude (in radians) of the transducer (e.g. pen stylus), in the range [0,π/2] - where 0 is parallel to the surface (X-Y plane), and π/2 is perpendicular to the surface. For hardware and platforms that do not report tilt or angle, the value MUST be 0.</p>
```

Comment:
I believe the default value of altitude should match the default value of tiltX/TiltY. So if they are 0 (for example when things aren't supported) then altitude should be pi/2.

One general suggestion is that if you add a new line after every comma or sentence it avoids the long lines here and easier to review. This is what I have seen in other specs as well and people seem to be generally happy with it.

Reply:
regarding the default value: this matches what touch events level 2 says https://w3c.github.io/touch-events/#dom-touch-altitudeangle

i've also tested this on an ipad (with no stylus, just using finger) and a modified version of my HUD tracker (that explicitly still shows touch events, rather than pointer events) https://patrickhlauke.github.io/touch/tracker/multi-touch-tracker-pointer-hud-toucheventsonly.html and indeed `azimuthAngle` and `altitudeAngle` are both defined (so the props exist), and return `0`. while i agree that it would be logical for the default to be pi/2, i think it's better to match touch events and what current implementations there do for defaults.

Reply:
as for line length, agree...but as most other text in our spec already breaks this idea, i just went with this approach at this point. may need a good separate isolated reformatting pull request

Reply:
I'm just trying to follow the same logic we did for pressure. In TouchEvent spec force is also zero if the device doesn't support it. But in PointerEvents it is 0.5 as we thought it makes more sense for the apps to still work with those devices.
With the same argument I believe we should diverge from TouchEvent spec in this case as well and set azimuth to pi/2 when it is not supported by the hardware. WDYT?

Reply:
sorry, only saw this comment after i went ahead with the merge...

while yes, we diverged for the pressure, i think here the danger / potential for confusion is greater because we went for exactly the same property names as touch events. happy to revisit though if you/others feel strongly about it.

Reply:
I feel like the ultimate question is how usually apps use this attribute. Since we do not expose a not-supported value for these apps will have no way of finding out whether 0 is real 0 or it is not supported. 

For pressure that was troublesome (and hence I'm not sure how that would even works in the touch event spec). If the app draws a line with a thickness relative to the pressure and with 0 pressure draws a line with 0 thickness (i.e. nothing) then if we had set the not supported pressure value to 0 then those app would just fail with those devices. Again they will have no way of finding out whether it is a real pressure or it is the not supported value. Hence the reason I'm confused how that choice even worked in TouchEvent spec. I can only assume all touch surfaces have some sort of pressure detection or apps just don't use the force attribute in TouchEvents the way I described above otherwise it has to fail.

And now the real question is how apps use the altitude value. Between 0 and pi/2 I felt pi/2 seems more natural to what apps might want to do with them. But I'm no expert. @smaug---- @grorg for more visibility.

Reply:
conceptually, i totally agree. just going by the reality of what webkit/safari seems to do with force and altitudeAngle in situations where those aren't detected (`0` in both cases). but yes, maybe @grorg has some insight into what, if anything, web apps on iOS (are supposed to) do here to distinguish between a "real" `0` and a "isn't supported" `0`)

Reply:
though i'll note that perhaps for `force` it's slightly moot because even the lightest of touches on a touchscreen is detected as having force `0`. it's only any further/harder force that is measured. and i believe that, since not all iOS devices have force sensing, the general approach taken by app developers has been to work regardless, of force / even when force is `0`, and then provide *additional* functionality based on any force greater than `0`.

i'm not completely against saying that in PE, we would instead default to an `altitudeAngle` of pi/2 for logical/conceptual reasons...but then we likely want to have a very clear warning.

further thought, and more crucially: IF in touch events the default is zero, and that's somehow due to what the sensor/OS sends along...then our conversion algorithm we provide also won't know if the zero it gets (when it then needs to convert it to `tiltX`/`tiltY`) is "not supported" or a "real" `0` ...

Reply:
I'm also okay either way as long as we clearly consider the use cases that these new attributes are used for and justify our decision.

Regarding the conversion within the UA in Chrome we actually do have a n/a for those values like pressure when things aren't supported and then we report 0 at the very end. So I would imagine we get the "not supported" from the platforms and we can do the right thing in the conversion phase.

Reply:
maybe `altitudeAngle` or `0` is simply assumed as being the default / not supported value because it's impossible (i assume) to really have that situation, since when the stylus is truly resting fully flat on the surface, the stylus tip isn't touching the screen itself so wouldn't be detectable anyway (unless you have a stylus that is "hover-capable" like, say, the pen on the galaxy note). i assume the apple pencil doesn't do hovering, in which case this assumption of "zero can't be the 'real' value" may make sense

Reply:
confirmed (as i don't have a device myself) here https://twitter.com/yatil/status/1253658517241765895

Reply:
I remember another reason for not exposing a "not supported" value was that developers didn't need to check for it every time. They just use the attribute and it works reasonably.
However, if we do go down the route of exposing a value that developers need to check for not supported case and do something other than what they would do with the actual value we are exposing I personally prefer to be much more explicit on that. Like expose an N/A or something that is clearly outside the range of possible values. I agree that Apple pen doesn't send 0 altitude at this point but I think we should be future proofing the API as well.

Reply:
question is if even for non-apple-pencil stylus scenarios if there's a situation where a completely flat stylus would be detected or not.

i'm just thinking that IF there are web apps already dealing with altitudeAngle for touch events/pencil, they'd be familiar with the "if it's zero, just ignore this altogether" scenario. and because we use the exact same property name in PE, it may cause more confusion / need for forking code if in PE we then have different default.

either way yes, we need to document this more clearly either way, and also look at the pseudocode to check for that edge case/scenario.

Reply:
split this out into https://github.com/w3c/pointerevents/issues/320